### PR TITLE
revised roadmap & README reorg

### PR DESCRIPTION
**Pull request by neuromusic**
reorganization of README
- moved content to docs
- added core team

revised roadmap
- high level roadmap with major tranches of near future work

### CODE REVIEW

**Comment by neuromusic** [comment]
fixes #605 

TODO: supported/experimental/unsupported definitions and modes

File: README.md [hunk 1]
```diff
@@ -6,218 +6,11 @@
 
 <img src="https://raw.githubusercontent.com/chanzuckerberg/cellxgene/master/docs/cellxgene-demo-1.gif" width="200" height="200" hspace="30"><img src="https://raw.githubusercontent.com/chanzuckerberg/cellxgene/master/docs/cellxgene-demo-2.gif" width="200" height="200" hspace="30"><img src="https://raw.githubusercontent.com/chanzuckerberg/cellxgene/master/docs/cellxgene-demo-3.gif" width="200" height="200" hspace="30">
```
Comment (csweaver) [inline README.md]: I realize we don't want too much duplication of documentation since it makes it difficult to update and know where to get info, however, I think we should have a bit more in the readme. I don't want to turn off users who are doing their first pass eval and want to know the key info without having to follow a chain of links.

File: README.md [hunk 2]
```diff
@@ -6,218 +6,11 @@
...
 
-## getting started
+Want to see where we are going? Check out [our roadmap](ROADMAP.md).
```
Comment (csweaver) [inline README.md]: Re: comment above -- Add the top line bullet points to this sections and then link out.

File: README.md [hunk 3]
```diff
@@ -6,218 +6,11 @@
...
-<summary> questions about algorithms </summary>
-
-<hr>
-
-> How are you computing and sorting differential expression results?
-
-Currently we use a [Welch's _t_-test](https://en.wikipedia.org/wiki/Welch%27s_t-test) implementation including the same variance overestimation correction as used in `scanpy`. We sort the `tscore` to identify the top N genes, and then filter to remove any that fall below a cutoff log fold change value, which can help remove spurious test results. The default threshold is `0.01` and can be changed using the option `--diffexp-lfc-cutoff`. We can explore adding support for other test types in the future.
-
-</details>
+If you want to learn how to install and use cellxgene, visit the [cellxgene docs](https://chanzuckerberg.github.io/cellxgene/).
```
Comment (csweaver) [inline README.md]: Re: comment above -- copy quick start guide here at the very least. Also I suggest a summary of what you can do with options ex. layout modes, etc.

File: README.md [hunk 4]
```diff
@@ -283,6 +68,14 @@ We are eager to explore integrations with other computational backends such as [
 
 Have questions, suggestions, or comments? You can come hang out with us by joining the [CZI Science Slack](https://join-cellxgene-users.herokuapp.com/) and posting in the `#cellxgene-users` channel. As mentioned above, please submit any feature requests or bugs as [Github issues](https://github.com/chanzuckerberg/cellxgene/issues). We'd love to hear from you!
```
Comment (csweaver) [inline README.md]: Can we change the above section to communication? And maybe  developer guide -> contributing? I think that matches what user expectations are for those sections more closely.

File: README.md [hunk 5]
```diff
@@ -283,6 +68,14 @@ We are eager to explore integrations with other computational backends such as [
...
 
+## core team
+
+- Colin Megill, frontend & product design
+- Charlotte Weaver, backend engineer
```
Comment (csweaver) [inline README.md]: can I be full stack engineer or software engineer instead?

File: ROADMAP.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
+# cellxgene roadmap
+
+We are very exited for _cellxgene_ to become a valuable tool in collaborations
+between computational biologists and experimental biologists working on
+single-cell transcriptomics data. _cellxgene_ is in active developmentis in
```
Comment (bkmartinjr) [inline ROADMAP.md]: typo "developmentis [sic]"

File: ROADMAP.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+active development, and we've love to include the community as we plan new
+features to work on.
+
+In the future, we plan to improve cellxgene in the ways described below.
```
Comment (bkmartinjr) [inline ROADMAP.md]: I would delete this sentence

File: ROADMAP.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
+# cellxgene roadmap
```
Comment (bkmartinjr) [inline ROADMAP.md]: Suggest the first para needs to give a pointer to where/how we would like input (eg, "if you have roadmap feedback, please submit as a github issue")

File: docs/getting-started.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
+## getting started
+
+You'll need **python 3.6+** and **Google Chrome**.
+The web UI is tested on OSX and Windows using Chrome, and the python CLI is tested on OSX and Ubuntu (via WSL/Windows). It should work on other platforms, but if you run into trouble let us know (see [help](#help-and-contact) below).
```
Comment (bkmartinjr) [inline docs/getting-started.md]: the help link is dead

File: README.md (hunk 3 above)
Comment (colinmegill) [reply README.md]: +1 quickstart at the top of the readme essential

File: ROADMAP.md (hunk 7 above)
Comment (colinmegill) [reply ROADMAP.md]: +1

File: README.md [hunk 10]
```diff
@@ -6,218 +6,11 @@
...
-- `--layout` to specify the layout as `tsne`, `umap`, `diffmap`, `phate`, `draw_graph_fa`, or `draw_graph_fr`
-- `--title` to show a title on the explorer
-- `--open` to automatically open the web browser after launching (OS X only)
-
-To see all options call
-
-```
-cellxgene launch --help
-```
-
```
[the remark this answers is not in the document]
Comment (colinmegill) [reply README.md]: (meaning, drop a link and say 'for more info')

File: README.md (hunk 3 above)
Comment (neuromusic) [reply README.md]: cool. on it.

File: README.md [hunk 11]
```diff
@@ -6,218 +6,11 @@
...
+Want to see where we are going? Check out [our roadmap](ROADMAP.md).
 
-You'll need **python 3.6+** and **Google Chrome**.
-The web UI is tested on OSX and Windows using Chrome, and the python CLI is tested on OSX and Ubuntu (via WSL/Windows). It should work on other platforms, but if you run into trouble let us know (see [help](#help-and-contact) below).
+## documentation
 
-To install run
-
-```
-pip install cellxgene
```
[the remark this answers is not in the document]
Comment (sidneymbell) [reply README.md]: Agreed. I'd leave all this in. Important to give people the quickstart for install + launch IMO.

File: README.md (hunk 4 above)
Comment (neuromusic) [reply README.md]: done.

File: README.md (hunk 3 above)
Comment (sidneymbell) [reply README.md]: +1. I don't think _all_ of this needs to stay, but I think it should contain:
1) quickstart how to install
2) quickstart how to launch
3) basics of data format (at least "h5ad with an embedding") and quickstart how to run `prepare` (i.e., that prepare exists, with an example command and pointing to the `--help` flag).

Comment (sidneymbell) [reply README.md]: (agreed the FAQ can move to the main docs page)

File: README.md (hunk 5 above)
Comment (neuromusic) [reply README.md]: yup

File: docs/index.md [hunk 12]
```diff
@@ -8,9 +8,9 @@ _cellxgene_ is an interactive data explorer for single-cell transcriptomics data
 #### Single-gene analyses (e.g. expression analysis)
 <img src="customGene.gif" width="600" />
 
-## getting started
+## quick start
 
-_cellxgene_ **only** supports Python 3.6. We recommend [installing _cellxgene_ into a conda or virtual environment.](/faq.html#how-do-i-create-a-python-36-environment-for-cellxgene)
+To install _cellxgene_ you need Python 3.6+. We recommend [installing _cellxgene_ into a conda or virtual environment.](/faq.html#how-do-i-create-a-python-36-environment-for-cellxgene)
```
Comment (csweaver) [inline docs/index.md]: dead link

File: README.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (sidneymbell) [reply README.md]: There's a similar level of detail currently in the "developer guide" section. Could we keep the high-level quickstart for both in, and move the details to the main docs for both?

File: docs/getting-started.md [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+pip install cellxgene[louvain]
+```
+
+If the aforementioned optional package installation fails, you can also install these packages directly:
+
+```
+pip install python-igraph louvain>=0.6
+```
+
+## conda and virtual environments
```
Comment (csweaver) [inline docs/getting-started.md]: do we want to explain what the difference is between the two? or have a why sentence?

**Comment by sidneymbell** [comment]
Added a few comments to the readme. The roadmap and getting started guides look great to me! Thanks, @neuromusic!

File: ROADMAP.md [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+*Last updated: April 11, 2019*
+
+## What we are building now
+
+In the near term, our goal is to enable teams of computational and experimental
+biologists to explore and annotate their single-cell RNA-seq data.
+
+There are 4 key features we plan to implement in the near term.
+
+- Click install and launch
```
Comment (csweaver) [inline ROADMAP.md]: Suggest: Click to install and launch or Simple install and launch

File: README.md [hunk 15]
```diff
@@ -6,220 +6,17 @@
 
 <img src="https://raw.githubusercontent.com/chanzuckerberg/cellxgene/master/docs/cellxgene-demo-1.gif" width="200" height="200" hspace="30"><img src="https://raw.githubusercontent.com/chanzuckerberg/cellxgene/master/docs/cellxgene-demo-2.gif" width="200" height="200" hspace="30"><img src="https://raw.githubusercontent.com/chanzuckerberg/cellxgene/master/docs/cellxgene-demo-3.gif" width="200" height="200" hspace="30">
 
-## getting started
+Want to see where we are going? Check out [our roadmap](ROADMAP.md).
 
-You'll need **python 3.6+** and **Google Chrome**.
-The web UI is tested on OSX and Windows using Chrome, and the python CLI is tested on OSX and Ubuntu (via WSL/Windows). It should work on other platforms, but if you run into trouble let us know (see [help](#help-and-contact) below).
+## Documentation
```
Comment (csweaver) [inline README.md]: We had been using all lowercase for headings to match with the all lowercase cellxgene. I personally liked this, but  understand if we want to go back to a more standard style guide.

Comment (csweaver) [reply README.md]: I would suggest adding the lowercase headings to the website too if we decide to keep this direction.

File: ROADMAP.md [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+
+We are very exited for _cellxgene_ to become a valuable tool in collaborations
+between computational biologists and experimental biologists working on
+single-cell transcriptomics data. _cellxgene_ is in active developmentis in
+active development, and we've love to include the community as we plan new
+features to work on.
+
+In the future, we plan to improve cellxgene in the ways described below.
+
+Please note: this roadmap may change at any time.
```
Comment (csweaver) [inline ROADMAP.md]:
```suggestion
Please note: this roadmap is subject to change.
```
It sounds a little less volatile which makes the roadmap seem a bit more useful.

File: ROADMAP.md [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+- Manual annotation workflows
+- Toggle embeddings
+- Gene information
+
+### Click install and launch
+
+The command line interface for installing and launching cellxgene is a barrier
+for users who are not used to Python or using the command line. We plan to
+support installation and launch of cellxgene on Mac and Windows. See
+https://github.com/chanzuckerberg/cellxgene/issues/687 for more details.
```
Comment (csweaver) [inline ROADMAP.md]: Could we pretty up the issue links a bit? ex. [Issue #697](https://github.com/chanzuckerberg/cellxgene/issues/687)

File: docs/getting-started.md [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+
+```
+cellxgene launch --help
+```
+
+There is an additional subcommand called `cellxgene prepare` that takes an existing dataset in one of several formats and applies minimal preprocessing and reformatting so that `launch` can use it (see [the next section](##data-formatting) for more info on `prepare`).
+
+## data formatting
+
+### assumptions
```
Comment (csweaver) [inline docs/getting-started.md]:
```suggestion
### requirements
```

File: README.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (neuromusic) [reply README.md]: yeah, I didn't want to touch the developer guide with this PR

File: docs/index.md (hunk 12 above)
Comment (neuromusic) [reply docs/index.md]: the link should work once the docs compile

Comment (csweaver) [reply docs/index.md]: I think you changed the section heading, though. Will that still work?

File: docs/getting-started.md (hunk 13 above)
Comment (neuromusic) [reply docs/getting-started.md]: perhaps. do you mind if we do a revision of this page's content in a separate PR?

File: README.md (hunk 15 above)
Comment (neuromusic) [reply README.md]: ah, gotcha

File: docs/index.md (hunk 12 above)
Comment (neuromusic) [reply docs/index.md]: Um... I think it's the same heading. Am I confused?

File: docs/getting-started.md [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+cellxgene launch dataset.h5ad --open
+```
+
+If you want an example dataset download [this file](https://github.com/chanzuckerberg/cellxgene/raw/master/example-dataset/pbmc3k.h5ad) and then call
+
+```
+cellxgene launch pbmc3k.h5ad --open
+```
+
+You should see your web browser open with the following
```
Comment (bkmartinjr) [inline docs/getting-started.md]: "If you are on MacOs, you should see your web browser ..."

File: README.md (hunk 1 above)
Comment (sidneymbell) [reply README.md]: Can you help me understand why you prefer to keep the developer guide in the readme but the user guide in the docs? I naively would have assumed the opposite :)

Comment (neuromusic) [reply README.md]: Sorry, let me rephrase.

Users should not have to go to the repo to find what they need. If users come to the repo, we should help them get to the docs ASAP. This is not necessarily true for developers.

We *can* expect developers to explore the repo to find what they need, so the README is an OK place to put the developer guide. However, since it bloats the README, I think there are better places to put it:
- CONTRIBUTING.md
- in the main docs

I think a good model here is Atom

Their "flight manual" (docs) has three distinct sections: getting started, using, hacking
https://flight-manual.atom.io/

The README takes a very light touch, basically acting as a landing page or table of contents for other resources:
https://github.com/atom/atom/blob/master/README.md

The contributing guide sets expectations for all types of contributions and links out to other resources where relevant.
https://github.com/atom/atom/blob/master/CONTRIBUTING.md

File: docs/getting-started.md (hunk 19 above)
Comment (neuromusic) [reply docs/getting-started.md]: This should work in Ubuntu, too (it does for jupyter, which also uses `webbrowser`). Is it just WSL where it breaks?

File: README.md (hunk 1 above)
Comment (csweaver) [reply README.md]: When I go to a readme for a project I want to be able to get a feel for the project and be able to download and use it if I like what I see. I personally get upset when I have to follow a series of links to get to the core info. 

To that point, I think developer docs can live outside the main readme. If I want to develop on the project that means that I've already used it in some capacity and am willing to invest more time.

However I disagree with the statement "If users come to the repo, we should help them get to the docs ASAP" They should be able to make an informed decision based on the readme whether they want to invest more of their time in the project (clicking links, reading docs, learning how to use options, etc.) No matter where we intercept them, the user should be able to get info on what cellxgene does, why they should use it over other options, how to get started initially, and how to get more info.

File: docs/getting-started.md (hunk 19 above)
Comment (bkmartinjr) [reply docs/getting-started.md]: Yes, likely just WSL due to the split-brain model...

File: docs/index.md [hunk 20]
```diff
@@ -25,13 +25,17 @@ curl -o pbmc3k.h5ad https://raw.githubusercontent.com/chanzuckerberg/cellxgene/m
...
+To explore more datasets already formatted for _cellxgene_, see [Data](data) or
+visit [Getting Started](getting-started) to learn more about formatting your own
+data for _cellxgene_. 
+
 ## getting help
 
 We'd love to hear from you!
 
-For questions, suggestions, or accolades, [join the `#cellxgene-users` channel on the CZI Science Slack](https://join-cziscience-slack.herokuapp.com/) and say "hi!".
+For questions, suggestions, or accolades, [join the `#cellxgene-users` channel on the CZI Science Slack](https://join-cellxgene-users.herokuapp.com/) and say "hi!".
```
Comment (csweaver) [inline docs/index.md]: did you update the app to work?

File: README.md [hunk 21]
```diff
@@ -2,224 +2,41 @@
...
+_cellxgene_ is an interactive data explorer for single-cell transcriptomics datasets, such as those coming from the [Human Cell Atlas](https://humancellatlas.org). Leveraging modern web development techniques to enable fast visualizations of at least 1 million cells, we hope to enable biologists and computational researchers to explore their data, and to demonstrate general, scalable, and reusable patterns for scientific data visualization.
 
 <img src="https://raw.githubusercontent.com/chanzuckerberg/cellxgene/master/docs/cellxgene-demo-1.gif" width="200" height="200" hspace="30"><img src="https://raw.githubusercontent.com/chanzuckerberg/cellxgene/master/docs/cellxgene-demo-2.gif" width="200" height="200" hspace="30"><img src="https://raw.githubusercontent.com/chanzuckerberg/cellxgene/master/docs/cellxgene-demo-3.gif" width="200" height="200" hspace="30">
 
-## getting started
+- Want to install and use cellxgene? Visit the [cellxgene docs](https://chanzuckerberg.github.io/cellxgene/).
+- Want to see where we are going? Check out [our roadmap](ROADMAP.md).
```
Comment (csweaver) [inline README.md]: I still feel strongly that we should add our top roadmap bullet points to the github readme as well as a statement of what we are trying to build (the collaboration tool)

File: README.md [hunk 22]
```diff
@@ -279,10 +83,18 @@ We have been working closely with the [`scanpy`](https://github.com/theislab/sca
 
 We are eager to explore integrations with other computational backends such as [`Seurat`](https://github.com/satijalab/seurat) or [`Bioconductor`](https://github.com/Bioconductor)
 
-## help and contact
+## communication
```
Comment (csweaver) [inline README.md]: Can we move this section up to after the quick start guide (and after the roadmap if we add that)? I think this is more important than developer communication (and overlaps slightly with that section).

File: docs/index.md (hunk 12 above)
Comment (csweaver) [reply docs/index.md]: I think that update goes with this PR actually.

File: docs/index.md [hunk 23]
```diff
@@ -8,9 +8,9 @@ _cellxgene_ is an interactive data explorer for single-cell transcriptomics data
 #### Single-gene analyses (e.g. expression analysis)
```
Comment (csweaver) [inline docs/index.md]: to lowercase, please

File: README.md (hunk 1 above)
Comment (sidneymbell) [reply README.md]: +1, @csweaver

File: docs/index.md (hunk 20 above)
Comment (neuromusic) [reply docs/index.md]: yep

File: README.md (hunk 22 above)
Comment (sidneymbell) [reply README.md]: I agree.

**Comment by sidneymbell** [comment]
These generally look great! Left a couple of small comments on the README, but none are blocking a merge IMO.

File: docs/index.md (hunk 12 above)
Comment (neuromusic) [reply docs/index.md]: perfect. fixed.

File: docs/getting-started.md (hunk 13 above)
Comment (neuromusic) [reply docs/getting-started.md]: tracking in #716

File: README.md (hunk 21 above)
Comment (neuromusic) [reply README.md]: done

File: README.md (hunk 22 above)
Comment (neuromusic) [reply README.md]: done. also renamed to "get in touch"

File: docs/index.md (hunk 23 above)
Comment (neuromusic) [reply docs/index.md]: fixed
